### PR TITLE
Chaos optimizations

### PR DESCRIPTION
**Pull request by sebader**
- Check if resource provider has been registered
- Some syntax cleanup
- Fail pipeline on failed chaos experiment

### CODE REVIEW

File: .ado/pipelines/templates/stages-full-release.yaml [hunk 1]
```diff
@@ -305,11 +305,11 @@ stages:
       parameters:
         stageName: 'deployChaosExperiments'
         customPrefix: '$(prefix)$(customSuffix)'
-        testDurationSeconds: '$(chaosExperimentDurationSeconds)' # Chaos test duration for each experiment
+        testDurationSeconds: '$(chaosExperimentDurationSeconds)'
```
Comment (msimecek) [inline .ado/pipelines/templates/stages-full-release.yaml:308]: I realize now that this isn't the most accurate name - we're not running chaos "tests". How about `experimentDurationSeconds` to indicate that **each** experiment will have this duration (regardless of running in parallel or sequentially)? Which is basically what the removed comment says.

File: .ado/pipelines/templates/stages-loadtest-azure.yaml [hunk 2]
```diff
@@ -190,8 +190,8 @@ stages:
           scriptType: pscore
           scriptLocation: inlineScript
           inlineScript: |
-            
-            Do {
+
+            do {
```
Comment (msimecek) [inline .ado/pipelines/templates/stages-loadtest-azure.yaml:194]: I still have an itch to rewrite all these scripts to "proper" PowerShell, but then I realize how many we have.... :D So this is ok I guess.

File: .ado/scripts/Invoke-ChaosMeshExperiment.ps1 [hunk 3]
```diff
@@ -88,5 +88,11 @@ do {
 }
 while (($statusResult.properties.status -ne "Success") -and ($statusResult.properties.status -ne "Failed"))
 
+if($statusResult.properties.status -eq "Failed")
```
Comment (msimecek) [inline .ado/scripts/Invoke-ChaosMeshExperiment.ps1:91]:
```suggestion
if ($statusResult.properties.status -eq "Failed")
```

File: src/testing/loadtest-azure/infra/main.tf [hunk 4]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "3.4.0"
+      version = "3.5.0"
```
Comment (msimecek) [inline src/testing/loadtest-azure/infra/main.tf:5]: Sneaky terraform update in chaos optimizations? ;)

File: .ado/pipelines/templates/stages-full-release.yaml (hunk 1 above)
Comment (sebader) [reply .ado/pipelines/templates/stages-full-release.yaml:308]: yep will change

Comment (sebader) [reply .ado/pipelines/templates/stages-full-release.yaml]: done

File: src/testing/loadtest-azure/infra/main.tf (hunk 4 above)
Comment (heoelri) [reply src/testing/loadtest-azure/infra/main.tf:5]: I wonder why this wasn't updated by dependabot in yesterday's PR #330

Comment (sebader) [reply src/testing/loadtest-azure/infra/main.tf:5]: it wasnt in the dependabot file then. I just updated that today

Comment (heoelri) [reply src/testing/loadtest-azure/infra/main.tf:5]: Ah - it's not dependabot's fault. Someone closed PR #390 ;)

Comment (sebader) [reply src/testing/loadtest-azure/infra/main.tf:5]: haha yeah since this PR will anyway bring it to main directly

**Review by msimecek — APPROVED** [review]

**Pull request merged**